wgengine/router/router_func.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package router
+
+import (
+	"tailscale.com/types/logger"
+)
+
+// setRoutesFunc applies the given router settings to the OS network stack.
+type setRoutesFunc func(cfg *Config) error
+
+// funcRouter passes the configs it receives to setRoutesFunc.
+// It is useful, for example, to pass configs into ipn-go-bridge on macOS/iOS.
+type funcRouter struct {
+	setRoutesFunc setRoutesFunc
+}
+
+// NewFuncRouter returns a Router which passes the configs it receives to setRoutesFunc.
+func NewFuncRouter(logf logger.Logf, setRoutesFunc setRoutesFunc) (Router, error) {
```

Comment:
Don't have exported symbols (NewFuncRouter) using unexported types (setRoutesFunc).  In godoc that'd be totally mysterious what the setRoutesFunc can even be.

But I'd just ditch the type and write the func signature in all two places.